ansible_base/api_documentation/customizations.py

```diff
@@ -0,0 +1,9 @@
+def apply_authentication_customizations() -> None:
+    """Make the schema generator of the type of DAB authentication classes"""
+    from drf_spectacular.authentication import SessionScheme
+
+    from ansible_base.authentication.session import SessionAuthentication
+
+    class MyAuthenticationScheme(SessionScheme):
+        target_class = SessionAuthentication  # full import path OR class ref
+        name = 'SessionAuthentication'
```

Comment:
So if authentication application is enabled we create this class but we don't seem to be doing anything with it? Does this get "magically" consumed by drf_spectacular or something? If so can we add a comment around that?

Reply:
In general the drf_spectacular library works like this - you have certain classes in imported modules, and it crawls them to build the schema.

But what you might be confused about here is that I only want this imported if _both_ the api_documentation and authentication apps are enabled, which is why this is called from the api_documentation ready method, gated on authentication being installed.

We could just define this below the `SessionAuthentication` class, and I think other places do that. But I think that's wrong, because `requirements_authentication.in` does not list drf_spectacular as a dependency, so it should be an import error if someone _only_ used authentication.

I would leave more of this in a comment, but I'm not sure if any of this is what your question is.

Reply:
added more docs